### ISSUE TITLE
Segmentation fault in AQL update

### ISSUE THREAD

**Issue by sajeevpr** [arangodb/arangodb#2455]
Dear team,
   We are getting segmentation fault while running the below query in Arangodb 3.1.18.

For u In U
  Filter u.NonExistField == null
 Let n = (u.k != null && u.k != "")?concat(u.k," ",u.l):concat(u.f," ",u.l)
Limit 50000
 update u._key WITH {"NonExistField": n} IN U

bt full gave me the below

Arangodb::basics:VelocityPackHelper::compare
AssocMulti
HashIndex::remove
deleteSecondaryIndexes
updateDocument
update
modifyLocal
update
work
ModificationBlock:getSome
execute
processQuery
createCursor
execute
executeEngine
asyncRun
run
runMe
startThread

Regards,
Sajeev

**Comment by fceller** [comment]
@sajeevpr is there a chance that you send us the core files? The backtrace helps to some degree, but with the cores it will be easier to find the issue. We are looking at the position in the source code that where given by the backtrace, but could not reproduce it yet.

**Comment by sajeevpr** [comment]
@fcellar,
    I sent the bt full output to Luca over email. Can you please collect it from him.

Regards,
Sajeev

**Comment by fceller** [comment]
@sajeevpr yes, I used the backtrace you send via email and order to reproduce the issue. Unfortunately, I did not manage to produce the crash. My hope was, that the core file you used to generate the backtrace give further insight. In principle, after installing the debug symbols it should be possible to extract more information (like line numbers and local variables).

Alternatively: Are these error reproducible on your side? If you execute the query with your dataset, does it always crash? If so, we could sign an NDA and test with your dataset.
